google_shopping_functions.js

```diff
@@ -15,16 +69,150 @@ var getItemsCount = function(itemData){
 
 //load products.json
 var data = require("./products.json")
+var items = data["items"];
+var brand = "Sony";
+var author = "Adorama Camera";
+var brand2 = "Nikon";
+var author2 = "eBay";
+
+
+
 
 //
 //USE FUNCTIONS HERE
 //
 
-
+var z = getItemsCount(items);
+var brandArray = getItemsByBrand(items, brand);
+var authorArray = getItemsByAuthor(items, author);
+var avaliableArray = getAvaliableProducts(items);
+var authorArray2 = getItemsByAuthor(items, author2);
+var brandArray2 = getItemsByBrand(items, brand2);
 
 
 
 
 //output item count using the getItemsCount function
-console.log('Item Count: ' + getItemsCount(data));
+console.log("Number of items "+ z);
+
```

Comment:
You actually don't need the below for loops here, because now you have functions to do it for you! You can call the functions you wrote above to output the information you need, like this:

```
// output items made by brand 'Sony'
var allSonyItems = getItemsByBrand(items, 'Sony');
```

Then, you can filter those results using your other function!

```
// output all available Sony items
getAvailableProducts(allSonyItems);
```